scalafix/rules/src/main/scala/rsc/rules/pretty/SemanticdbPrinter.scala

```diff
@@ -81,6 +82,43 @@ class SemanticdbPrinter(env: Env, index: DocumentIndex, config: RscCompatConfig)
     case _ => sys.error(s"unsupported tree $tree")
   }
 
+  def symbolOf(signature: Signature): Option[String] =
+    signature match {
+      case TypeSignature(_, lower: TypeRef, upper: TypeRef) if lower == upper => Some(lower.symbol)
+      case _ => None
+    }
+
+
+  def symIsInScope(sym: String, env: Env): Boolean = {
+    extractName(sym)
+      .map(env.lookup) match {
+      // TODO: If the lookup returns NoSymbol, we can insert an import and skip the prefix.
+      // The logic to implement this is left for future work.
+      case None => false
+      case Some(symFromScope) => isSameOrAlias(sym, symFromScope)
+    }
+  }
+
+  def isSameOrAlias(sym: String, envSym: String) =
```

Comment:
What does `envSym` mean?

Reply:
Could just as well have been sum' or sym2, but it means the sum pulled from env.lookup. Suggestions for a better name?

Reply:
`sym2` doesn't work well in my opinion, because `sym` and `sym2` are not interchangeable - only `sym2` is dealiased, but not `sym`.

Reply:
I can't come up with a better name at the moment. In situations like that, I inline helpers into their callsites, which saves the mental effort of naming.

Reply:
I prefer not to inline here because I lose the benefit of naming, isSameOrAlias. I will add a doc comment for clarity.